client/src/main/java/io/pravega/client/connection/impl/RawClient.java

```diff
@@ -98,23 +98,25 @@ public void authTokenCheckFailed(WireCommands.AuthTokenCheckFailed authTokenChec
 
     public RawClient(PravegaNodeUri uri, ConnectionPool connectionPool) {
         this.segmentId = null;
-        this.connection = connectionPool.getClientConnection(flow, uri, responseProcessor)
-        .exceptionally(e -> {
+        this.connection = new CompletableFuture<>();
+        connectionPool.getClientConnection(flow, uri, responseProcessor, this.connection);
+        this.connection.exceptionally(e -> {
```

Comment:
This is different from previous code since `this.connection` will not be completed with `ConnectionFailedException(e)`

```
//previous snip.
this.connection = connectionPool.getClientConnection(flow, uri, responseProcessor)
        this.connection = new CompletableFuture<>();
        .exceptionally(e -> {
```

Reply:
The CompletableFuture will not be failed with ConnectionFailedException and it is a change in behaviour for the retry logic

Reply:
Thanks! That's a good point!